core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -374,6 +372,21 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
     zkClient.getClusterId.getOrElse(zkClient.createOrGetClusterId(CoreUtils.generateUuidAsBase64))
   }
 
+  private def registerBroker(zkClient: KafkaZkClient): BrokerInfo = {
+    if (!config.advertisedListenersEndPoints.isEmpty) {
+      val hostNames = config.advertisedListenersEndPoints.map(_.host)
+      zkClient.getAllBrokersInCluster.foreach(broker => {
+          val commonHosts = broker.endPoints.map(_.host).intersect(hostNames)
```

Comment:
Should we check port as well? I think using the same host is pretty common, especially for development.

Reply:
Yes, we can check port also. Updated the code.